mzLib/MassSpectrometry/MzSpectra/SpectralSimilarity.cs

```diff
@@ -315,6 +319,37 @@ private double[] NormalizeSpectrumSum(double[] spectrum)
             return sum;
         }
 
+        // This method should only be used with the SpectrumSum normalization method
+        // This method should only be used when allPeaks is set to true
+        public double? SpectralEntropy()
+        {
+            if (_intensityPairs.First().Item1 == -1)
```

Comment:
can delete this first check. add comment that the test for some valid pairs is performed higher in the food chain